sdm/src/main/java/com/sap/cds/sdm/handler/applicationservice/SDMUpdateAttachmentsHandler.java

```diff
@@ -0,0 +1,140 @@
+package com.sap.cds.sdm.handler.applicationservice;
+
+import static com.sap.cds.sdm.persistence.DBQuery.*;
+
+import com.sap.cds.CdsData;
+import com.sap.cds.Result;
+import com.sap.cds.reflect.CdsEntity;
+import com.sap.cds.sdm.handler.TokenHandler;
+import com.sap.cds.sdm.model.SDMCredentials;
+import com.sap.cds.sdm.persistence.DBQuery;
+import com.sap.cds.sdm.service.SDMService;
+import com.sap.cds.services.authentication.AuthenticationInfo;
+import com.sap.cds.services.authentication.JwtTokenAuthenticationInfo;
+import com.sap.cds.services.cds.ApplicationService;
+import com.sap.cds.services.cds.CdsUpdateEventContext;
+import com.sap.cds.services.handler.EventHandler;
+import com.sap.cds.services.handler.annotations.Before;
+import com.sap.cds.services.handler.annotations.HandlerOrder;
+import com.sap.cds.services.handler.annotations.ServiceName;
+import com.sap.cds.services.persistence.PersistenceService;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@ServiceName(value = "*", type = ApplicationService.class)
+public class SDMUpdateAttachmentsHandler implements EventHandler {
+
+  private final PersistenceService persistenceService;
+  private final SDMService sdmService;
+
+  public SDMUpdateAttachmentsHandler(PersistenceService persistenceService, SDMService sdmService) {
+    this.persistenceService = persistenceService;
+    this.sdmService = sdmService;
+  }
+
+  // @Before(event = CqnService.EVENT_UPDATE)
+  @Before
+  @HandlerOrder(HandlerOrder.EARLY)
+  public void processBefore(CdsUpdateEventContext context, List<CdsData> data) {
+    try {
+      rename(context, data);
+    } catch (IOException e) {
+      context.getMessages().error("Error renaming attachment");
+    }
+  }
+
+  public void rename(CdsUpdateEventContext context, List<CdsData> data) throws IOException {
```

Comment:
can we somehow avoid this boiler plate code

Reply:
Added isFileNameDuplicateInDrafts() to SDMUtils.java, as same code was being used in both SDMCreateAttachmentsHandler & SDMUpdateAttachmentsHandler